Mock Data & Endpoint

sanchay811 opened the pull request:
- Added scripts for running the mock endpoints using json-server and for generating schema data 
  - Do this by running npm run start-mockapi and to generate fresh data use npm run generate-mock-data
  - Mocked endpoints will run at localhost:3000/<endpoint name>
    - Current endpoints: /users and /venues
- Updated dependencies
- Removed redundant files
- Created template for schema, located at src\schema\schema.js
- Added random data generator script (scripts\generateMockData.js)

howardhenry commented on package.json:
```diff
@@ -8,7 +8,9 @@
     "test": "echo \"Error: no test specified\" && exit 1",
-    "precommit": "npm run eslint && npm run sass-lint"
+    "precommit": "npm run eslint && npm run sass-lint",
+    "generate-mock-data": "node scripts/generateMockData",
```
Use consistent file naming conventions. The JS community typically uses dash-case for filenames

sanchay811 commented:
@howardhenry Can you please review for the following updates:
- Removed mock server files
- Created reusable routes and constants files
- Expanded on reusable Text-Input component
- Introduced FireBase integration, simple sign up setup
- Added additional dummy route points
 
**Steps to test SignUp**
1. Go to https://firebase.google.com/ and create a free account, you will have to setup a sample project to connect to this app. Then once a project is setup click on Authentication under Develop in the side navbar on the firebase-console page. Under Sign-in method you want to enable email/password. 
2. After checking out this branch, create your own .env file in the following format and update with your firebase config values: 

*File Location: src\components\firebase\firebase.js*
```javascript
API_KEY="YOUR_KEY"
AUTH_DOMAIN="YOUR_AUTH_DOMAIN"
DATABASE_URL="YOUR_DATABASE_URL"
PROJECT_ID="YOUR_PROJECT_ID"
STORAGE_BUCKET="YOUR_STORAGE_BUCKET"
MESSAGE_SENDER_ID="YOUR_MESSAGE_SENDER_ID"`

Your config information can be found once logged into firebase -> have a sample project setup -> in the firebase console, next `project overview` the settings icon -> project settings -> scroll down and click on the icon with opening and closing tag, you will see your config here.

3. Start the app
4. Navigate to end-point `localhost:8080/sign-up`
5. Fill in details and submit form
6. Check your firebase under Authentication.

howardhenry commented:
Hey @sanchay811, it looks like this PR was created a month ago and was already approved then. However, you never merged it. Instead, you added more changes and it is ballooning out of size. You need to create you changes in digestible chunks. So, please merge the changes that were made _before_ the approval and create a new PR with the subsequent changes.

sanchay811 commented:
@howardhenry Yeah you're right it is getting out of control, will update it.
